Blockzilla/URLBar.swift

```diff
@@ -675,6 +675,7 @@ class URLBar: UIView {
 
     @objc private func didPressClear() {
         urlText.text = nil
+        userInputText = nil
```

Comment:
### Adding the debouncer created a race condition

if a user was typing the same letter (or holding backspace) and very quickly pressed the 'clear' button, the query would clear but would then go back to being the text right before the clear button. This was because we weren't setting userInputText to nil, which is what's checked when we [check if the callback is stale](https://github.com/mozilla-mobile/focus-ios/pull/1574/files#diff-93f8e8b1353c1dabb447526fe4178103R832).